### ISSUE TITLE
Version incorrectly reported (0.0.0)

### ISSUE THREAD

**Issue by basnijholt** [conda-forge/h5netcdf-feedstock#42]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

```bash
micromamba create -n h5netcdf-version h5netcdf
micromamba activate h5netcdf-version
pip list
```
prints
```bash
❯ pip list
Package         Version
--------------- -------
cached-property 1.5.2
h5netcdf        0.0.0
h5py            3.8.0
numpy           1.24.2
packaging       23.0
pip             23.0.1
setuptools      67.4.0
wheel           0.38.4
```

The `__version__` string is correctly set:
```bash
❯ python -c "import h5netcdf; print(h5netcdf.__version__)"
1.1.0
```

When creating an new env and pip installing h5netcdf, the version is fine:
```
❯ pip list | grep h5netcdf
h5netcdf   1.1.0
```

### Installed packages

```shell
❯ conda list
List of packages in environment: "/Users/basnijholt/micromamba/envs/py310-test"

  Name             Version       Build                    Channel
───────────────────────────────────────────────────────────────────────
  bzip2            1.0.8         h3422bc3_4               conda-forge
  c-ares           1.18.1        h3422bc3_0               conda-forge
  ca-certificates  2022.12.7     h4653dfc_0               conda-forge
  cached-property  1.5.2         hd8ed1ab_1               conda-forge
  cached_property  1.5.2         pyha770c72_1             conda-forge
  h5netcdf         1.1.0         pyhd8ed1ab_0             conda-forge
  h5py             3.8.0         nompi_py311ha096f4f_101  conda-forge
  hdf5             1.14.0        nompi_h6b85c65_102       conda-forge
  krb5             1.20.1        h69eda48_0               conda-forge
  libaec           1.0.6         hb7217d7_1               conda-forge
  libblas          3.9.0         16_osxarm64_openblas     conda-forge
  libcblas         3.9.0         16_osxarm64_openblas     conda-forge
  libcurl          7.88.1        h9049daf_0               conda-forge
  libcxx           15.0.7        h75e25f2_0               conda-forge
  libedit          3.1.20191231  hc8eb9b7_2               conda-forge
  libev            4.33          h642e427_1               conda-forge
  libffi           3.4.2         h3422bc3_5               conda-forge
  libgfortran      5.0.0         11_3_0_hd922786_28       conda-forge
  libgfortran5     11.3.0        hdaf2cc0_28              conda-forge
  liblapack        3.9.0         16_osxarm64_openblas     conda-forge
  libnghttp2       1.51.0        hae82a92_0               conda-forge
  libopenblas      0.3.21        openmp_hc731615_3        conda-forge
  libsqlite        3.40.0        h76d750c_0               conda-forge
  libssh2          1.10.0        h7a5bd25_3               conda-forge
  libzlib          1.2.13        h03a7124_4               conda-forge
  llvm-openmp      15.0.7        h7cfbb63_0               conda-forge
  ncurses          6.3           h07bb92c_1               conda-forge
  numpy            1.24.2        py311h60f8152_0          conda-forge
  openssl          3.0.8         h03a7124_0               conda-forge
  packaging        23.0          pyhd8ed1ab_0             conda-forge
  pip              23.0.1        pyhd8ed1ab_0             conda-forge
  python           3.11.0        h3ba56d0_1_cpython       conda-forge
  python_abi       3.11          3_cp311                  conda-forge
  readline         8.1.2         h46ed386_0               conda-forge
  setuptools       67.4.0        pyhd8ed1ab_0             conda-forge
  tk               8.6.12        he1e0b03_0               conda-forge
  tzdata           2022g         h191b570_0               conda-forge
  wheel            0.38.4        pyhd8ed1ab_0             conda-forge
  xz               5.2.6         h57fd34a_0               conda-forge
```

### Environment info

```shell
❯ conda info

                                           __
          __  ______ ___  ____ _____ ___  / /_  ____ _
         / / / / __ `__ \/ __ `/ __ `__ \/ __ \/ __ `/
        / /_/ / / / / / / /_/ / / / / / / /_/ / /_/ /
       / .___/_/ /_/ /_/\__,_/_/ /_/ /_/_.___/\__,_/
      /_/


            environment : py310-test (active)
           env location : /Users/basnijholt/micromamba/envs/py310-test
      user config files : /Users/basnijholt/.mambarc
 populated config files : /Users/basnijholt/.condarc
       libmamba version : 1.3.0
     micromamba version : 1.3.0
           curl version : libcurl/7.86.0 SecureTransport (OpenSSL/3.0.7) zlib/1.2.13 libssh2/1.10.0 nghttp2/1.47.0
     libarchive version : libarchive 3.6.2 zlib/1.2.13 bz2lib/1.0.8 libzstd/1.5.2
       virtual packages : __unix=0=0
                          __osx=13.2.1=0
                          __archspec=1=arm64
               channels : https://conda.anaconda.org/conda-forge/osx-arm64
                          https://conda.anaconda.org/conda-forge/noarch
       base environment : /Users/basnijholt/micromamba
               platform : osx-arm64
```

**Comment by hmaarrfk** [comment]
There is no version in their setup.py file.
https://github.com/h5netcdf/h5netcdf/blob/main/setup.py#L38

report upstream?

**Comment by basnijholt** [comment]
I opened https://github.com/h5netcdf/h5netcdf/pull/208

**Comment by hmaarrfk** [comment]
since you have opened a 1 liner PR, i would accept it if `pip list` shows the versiona appearing in the azure pipelines

**Comment by kmuehlbauer** [comment]
I've tried to replicate that locally, but failed.

I'm wondering if that can be solved by adding `setuptools_scm` to host-section.

**Comment by kmuehlbauer** [comment]
See #44

**Comment by kmuehlbauer** [comment]
@basnijholt Thanks for reporting and diving into this. The issue should be resolved now by #44.
